tools/c7n_tencentcloud/c7n_tencentcloud/actions/tags.py

```diff
@@ -1,11 +1,322 @@
+        for t in tags:
+            if t["Key"] == old_key:
+                return t
+        return None
+
+    def _get_rename_request_params(self, resource):
+        """
+        get rename request params
+        https://cloud.tencent.com/document/api/651/35322
+        """
+        param = {}
+        old_tag = self._get_tag_params(resource)
+        qcs_list = self.manager.source.get_resource_qcs([resource])
+        param.update({"Resource": qcs_list[0]})
+        if old_tag is not None:
+            replaceTags = {"ReplaceTags": [{
+                "TagKey": self.data.get("new_key"),
+                "TagValue": old_tag["Value"]
+            }]}
+            param.update(replaceTags)
+
+        deleteTags = {"DeleteTags": [{"TagKey": self.data.get('old_key')}]}
+        param.update(deleteTags)
+
+        return param
+
+    def process(self, resources, event=None):
+        """
+        process rename tag
+        """
+        try:
+            client = self.get_client()
+            for resource in resources:
+                params = self._get_rename_request_params(resource)
+                resp = client.execute_query(self.t_api_method_name, params)
+                self.log.debug("%s , params: %s,resp: %s ", self.data.get('type'),
+                               json.dumps(params), json.dumps(resp))
+        except (RetryError, TencentCloudSDKException) as err:
+            raise PolicyExecutionError(err) from err
+
+
+class DeleteTagAction(TagAction):
+    """Delete Tag"""
+    schema = type_schema("remove-tag",
+                         tag={"type": "string"},
+                         tags={"type": "array"},
+                         msg={"type": "string"})
+    schema_alias = True
+    t_api_method_name = "UnTagResources"
+
+    def __init__(self, data=None, manager=None, log_dir=None):
+        super().__init__(data, manager, log_dir)
+        self.tag_request_params = self.data.get("tags")
+
+    def validate(self):
+        """validate"""
+        if not self.data.get('tags'):
+            raise FilterValidationError("Must specify tags")
+        return self
+
+
+class TagDelayedAction(TagAction):
+    """
+    Tag resources for future action.
+    """
+    schema = type_schema(
+        "mark-for-op",
+        tag={"type": "string"},
+        msg={"type": "string"},
+        days={"type": "number", "minimum": 0},
+        hours={"type": "number", "minimum": 0},
+        # https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+        # TZ database name
+        tz={"type": "string"},
+        op={"type": "string"})
+    schema_alias = True
+    t_api_method_name = "TagResources"
+
+    default_template = "Resource does not meet policy: {op}@{action_date}"
+
+    def __init__(self, data=None, manager=None, log_dir=None):
+        super().__init__(data, manager, log_dir)
+        self.resource_type = self.manager.get_model()
+        try:
+            self.tz = pytz.timezone(self.data.get("tz", "utc"))
```

Comment:
nit: this should move to validate, we guarantee validate is called prior to processing any resources.